pkg/asset/installconfig/aws/session.go

```diff
@@ -93,14 +94,31 @@ func GetSessionWithOptions(optFuncs ...SessionOptions) (*session.Session, error)
 			return nil, err
 		}
 	}
+	err = validate(ssn)
```

Comment:
This validation should be done as part of the platform creds check rather than on every create of a session.
https://github.com/openshift/installer/blob/6b2861cd4bead8e8e290b51fab8187d3a0d1abb5/pkg/asset/installconfig/platformcredscheck.go#L47-L51

Reply:
Understood, I will try to rework that.

Reply:
Hey @staebler,

I attempted to move the validation to the credentials validation, but it does not seem to work, the installer still hangs. Here's the diff:

```
diff --git a/pkg/asset/installconfig/platformcredscheck.go b/pkg/asset/installconfig/platformcredscheck.go
index 2fedfeb9e..5dd3a98ca 100644
--- a/pkg/asset/installconfig/platformcredscheck.go
+++ b/pkg/asset/installconfig/platformcredscheck.go
@@ -3,9 +3,11 @@ package installconfig
 import (
        "context"
        "fmt"
+       "github.com/aws/aws-sdk-go/service/sts"

        "github.com/pkg/errors"

+       awssdk "github.com/aws/aws-sdk-go/aws"
        "github.com/openshift/installer/pkg/asset"
        gcpconfig "github.com/openshift/installer/pkg/asset/installconfig/gcp"
        openstackconfig "github.com/openshift/installer/pkg/asset/installconfig/openstack"
@@ -45,7 +47,14 @@ func (a *PlatformCredsCheck) Generate(dependencies asset.Parents) error {
        platform := ic.Config.Platform.Name()
        switch platform {
        case aws.Name:
-               _, err := ic.AWS.Session(ctx)
+               ssn, err := ic.AWS.Session(ctx)
+               if err != nil {
+                       return err
+               }
+               quickSsn := ssn.Copy(&awssdk.Config{MaxRetries: awssdk.Int(3)})
+               svc := sts.New(quickSsn)
+               input := &sts.GetCallerIdentityInput{}
+               _, err = svc.GetCallerIdentity(input)
                if err != nil {
                        return err
                }
```

The installer hangs at this point:

```
time="2020-11-02T11:31:46+01:00" level=debug msg="listing AWS hosted zones"
```

This specific message appears as part of the install config which is called before the credentials verification. Any advice on how to proceed from here?

Reply:
Ah, for the path where the installer is prompting the user for input, the call to collect the hosted zones happens before the creds check asset is generated. I suggest that you add a function in pkg/asset/installconfig/aws/session.go that calls `GetCallerIdentityInput` using a passed-in session. Call the new function from the platformcredscheck asset and also immediately after getting the session that is used to collect AWS-specific configuration for the install-config [1].

[1] https://github.com/openshift/installer/blob/a409de073b472944be458283e69ef831cbdfd1c9/pkg/asset/installconfig/aws/platform.go#L33-L36

Reply:
This seems to do the trick, here's the new user experience:

```
$ ./bin/openshift-install create cluster
? Platform aws
INFO Credentials loaded from the "default" profile in file "/home/janoszen/.aws/credentials"
FATAL failed to fetch Metadata: failed to fetch dependency of "Metadata": failed to fetch dependency of "Cluster ID": failed to fetch dependency of "Install Config": failed to fetch dependency of "Base Domain": failed to generate asset "Platform": ExpiredToken: The security token included in the request is expired
FATAL   status code: 403, request id: 06889a22-1ba9-403c-ac0d-260d1c6a3cc6
```